src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -6651,11 +6673,25 @@ and TcCtorCall isNaked cenv env tpenv (overallTy: OverallTy) objTy mObjTyOpt ite
         error(Error((if superInit then FSComp.SR.tcInheritCannotBeUsedOnInterfaceType() else FSComp.SR.tcNewCannotBeUsedOnInterfaceType()), mWholeCall))
 
     match item, args with
-    | Item.CtorGroup(methodName, minfos), _ ->
+    | Item.CtorGroup(methodName, minfos), argExprs ->
         let meths = List.map (fun minfo -> minfo, None) minfos
         if isNaked && TypeFeasiblySubsumesType 0 g cenv.amap mWholeCall g.system_IDisposable_ty NoCoerce objTy then
             warning(Error(FSComp.SR.tcIDisposableTypeShouldUseNew(), mWholeCall))
 
+        let ctorArgs = GetConstructorArgsIdents argExprs
+        
+        if not ctorArgs.IsEmpty then
+            // Here we will have the ValRefs(Members, Properties) from F# types
+            let valRefs =
+                minfos
+                |> List.collect (fun minfo-> minfo.ApparentEnclosingTyconRef.MembersOfFSharpTyconSorted)
```

Comment:
This is the part I would ideally like to get rid of since you now also have the approach with finalAssignedItemSetters - avoiding to collect (minfos * #props in a type).

Reply:
Looked in more detail and, Yeah we can completely remove this. See my last commit